core/src/main/scala/org/apache/spark/sql/execution/columnar/ColumnInsertExec.scala

```diff
@@ -78,7 +82,134 @@ case class ColumnInsertExec(_child: SparkPlan, partitionColumns: Seq[String],
     else ("1", "0x0")
   }
 
+  def loop(code: String, numTimes: Int): String = {
+    s"""
+      for(int i=0;i< $numTimes; i++){
+          $code
+      }""".stripMargin
+  }
```

Comment:
To be paranoid, the loop code can be a separate method itself for many of the calls I saw below. But will need the variables to be either members or as arguments to the method. So you can have two versions of loop() method -- one which creates in a separate method where this is simple enough to do, and this one.